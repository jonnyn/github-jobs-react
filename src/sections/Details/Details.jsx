import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from "react-router";
import { Container, Row, Col, Media } from 'react-bootstrap';
import { fetchJobById } from 'redux/jobsSlice';
import { AppSpinner } from 'components';

class Details extends Component {
    constructor(props) {
        super(props);
        this.state = {
            isLoading: true
        }
    }

    async componentDidMount() {
        await this.props.dispatch(fetchJobById(this.props.match.params.jobId));
        this.setState({isLoading: false});
    }

    render() {
        if (this.state.isLoading || !this.props.activeJob) return <AppSpinner />

        const { type, created_at, company, company_url, location, title, description, company_logo} = this.props.activeJob;
        return (
            <Container fluid="md">
                <Row>
                    <Col md={4}>
                        <Media>
                            <img
                                width={64}
                                height={64}
                                className="mr-3"
                                src={company_logo}
                                alt="Company logo"
                            />
                            <Media.Body>
                                <h3>{company}</h3>
                                <p>{company_url}</p>
                                <p>{location}</p>
                            </Media.Body>
                        </Media>
                    </Col>
                </Row>
                <Row md={4}>
                    <Col md={{ span: 6, offset: 1 }}>
                        <h4>{title}</h4>
                        <h5>{type}</h5>
                        <br />
                        <p dangerouslySetInnerHTML={{__html: description}} />
                        <br />
                        <p>{created_at}</p>
                    </Col>
                </Row>
            </Container>
        );
    }
};

const mapStateToProps = state => ({
    activeJob: state.jobs.activeJob
});

const mapDispatchToProps = dispatch => ({
    dispatch
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Details));