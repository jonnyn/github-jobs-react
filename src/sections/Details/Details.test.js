import React from 'react';
import { shallow } from 'enzyme';
import { Details } from './Details';

const setup = () => {
    const props = {
        dispatch: jest.fn(),
        match: {params: {jobId: 'fakeId'}}
    }
    const enzymeWrapper = shallow(<Details {...props} />)

    return {
        props,
        enzymeWrapper
    }
};

it('renders Details without crashing', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.exists()).toBe(true);
});

it('renders correctly loading state', () => { 
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper).toMatchSnapshot();
});