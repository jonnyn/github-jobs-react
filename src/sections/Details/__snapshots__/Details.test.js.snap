// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly loading state 1`] = `ShallowWrapper {}`;
